File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        complements = dict()

        for i, c in enumerate(nums):
            if c not in complements:
                complements[target - c] = i
            else:
                return (i, complements[c])


        

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        std::unordered_map<int, int> complements;

        for (int i = 0; i < nums.size(); i++){
            int num = nums[i];
            auto complement_index = complements.find(num);
            if (complement_index != complements.end()){
                return {i, complement_index->second};
            }
            complements[target - num] = i;
        }
        
        auto dummy = {-1, 1};
        return dummy;

    }
};